src/miniscript/astelem.rs

```diff
@@ -146,11 +147,13 @@ impl<Pk: MiniscriptKey> Terminal<Pk> {
                 let keys: Result<Vec<Q>, _> = keys.iter().map(&mut *translatefpk).collect();
                 Terminal::Multi(k, keys?)
             }
-        })
+        };
+        Ctx::check_frag_validity(&frag).expect("Translate failed!");
```

Comment:
I am not completely sure how to handle this error. There is already one error type because of the Function argument. Secondly, we would also another because the translation may not be valid(for ex: translating string to uncompressed pubkeys in segwit descriptor).

Reply:
We can 
1. Create a enum sum type for both errors
2. Return a Result<Result<ms, err1>,err2>
3. Panic!